File: app/logic/js/logic/colorname.test.js
const testsN = [
    {
        name: "happy path: Aero",
        in: {r: 124, g: 185, b: 232},
        want: "Aero",
    },
    {
        name: "happy path: Black",
        in: {r: 0, g: 0, b: 0},
        want: "Black",
    },
    {
        name: "happy path: White",
        in: {r: 255, g: 255, b: 255},
        want: "White",
    },
]

testsN.forEach((test) => {
    const got = N(test.in.r, test.in.g, test.in.b);
    if (got !== test.want) {
        throw Error(`Test "${test.name}". got: ${got}, want: ${test.want}`);
    }
});

File: app/logic/go/logic/colorname_test.go
package logic

import "testing"

func TestColorName(t *testing.T) {
	type args struct {
		r, g, b uint8
	}
	tests := []struct {
		name string
		args args
		want string
	}{
		{
			name: "exact name",
			args: args{
				r: 124,
				g: 185,
				b: 232,
			},
			want: "Aero",
		},
		{
			name: "black",
			args: args{
				r: 0,
				g: 0,
				b: 0,
			},
			want: "Black",
		},
		{
			name: "white",
			args: args{
				r: 255,
				g: 255,
				b: 255,
			},
			want: "White",
		},
	}
	for _, tt := range tests {
		t.Run(
			tt.name, func(t *testing.T) {
				if got := Name(tt.args.r, tt.args.g, tt.args.b); got != tt.want {
					t.Errorf("Name() = %v, want %v", got, tt.want)
				}
			},
		)
	}
}
